packages/pigeon/lib/swift_generator.dart

```diff
@@ -125,11 +125,31 @@ class SwiftGenerator extends StructuredGenerator<SwiftOptions> {
         indent, classDefinition.documentationComments, _docCommentSpec,
         generatorComments: generatedComments);
 
-    indent.write('struct ${classDefinition.name} ');
+    if (classDefinition.isSwiftClass) {
+      String inheritNSObject = '';
+      if (classDefinition.isSwiftObjcInteropClass) {
+        inheritNSObject = ': NSObject';
+        indent.writeln('@objc');
```

Comment:
iirc `@objc` isn't necessary since it's already a subclass of `NSObject`, which is already exposed to objc runtime.

Reply:
Oh in addition to above, we probably need to add `@objc` for all custom members that are not inherited from `NSObject` (basically all the properties and the `init` function). 

Can you also write some ObjC tests to make sure it works?

Reply:
all properties? even primitives? or only custom types?

Reply:
All properties and methods that are not inherited from `NSObject` need to be annotated, in order for them to be accessibly by objc runtime. The class definition itself and its members inherited from `NSObject` are already implicitly annotated.

Reply:
So yes, even the properties of primitive types.